src/node_native_module.cc

```diff
@@ -19,100 +19,113 @@ using v8::Isolate;
 using v8::Local;
 using v8::Maybe;
 using v8::MaybeLocal;
+using v8::Name;
 using v8::Object;
+using v8::PropertyCallbackInfo;
 using v8::Script;
 using v8::ScriptCompiler;
 using v8::ScriptOrigin;
 using v8::Set;
 using v8::String;
-using v8::TryCatch;
 using v8::Uint8Array;
 using v8::Value;
 
-void NativeModule::GetNatives(Environment* env, Local<Object> exports) {
-  DefineJavaScript(env, exports);
+Local<Object> MapToObject(Local<Context> context,
+                          const NativeModuleRecordMap& in) {
+  Isolate* isolate = context->GetIsolate();
+  Local<Object> out = Object::New(isolate);
+  for (auto const& x : in) {
+    Local<String> key = OneByteString(isolate, x.first.c_str(), x.first.size());
+    out->Set(context, key, x.second.ToStringChecked(isolate)).FromJust();
+  }
+  return out;
+}
+
+Local<Set> ToJsSet(Local<Context> context, const std::set<std::string>& in) {
```

Comment:
Hm, maybe this one could better be implemented as an `ToV8Value()` overload from `util.h`? It fits in with the others pretty well, I think (the downside being that we always use UTF8 there unconditionally…)

Reply:
@addaleax The current implementation is only capable of converting std::set<const std::string> to an array...not exactly what we want here, but I can leave a TODO